controllers/watcher_controller.go

```diff
@@ -245,6 +253,24 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 		return ctrl.Result{}, errors.New("error retrieving required data from transporturl secret")
 	}
 
+	// Prometheus config secret
+
+	hashPrometheus, _, _, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.PrometheusSecret},
+		[]string{
+			PrometheusHost,
+			PrometheusPort,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		r.RequeueTimeout,
+	)
+	if err != nil || hashPrometheus == "" {
+		// Empty hash means that there is some problem retrieving the key from the secret
+		return ctrl.Result{}, errors.New("error retrieving required data from prometheus secret")
+	}
+
```

Comment:
by the way we need to create a finalise on the PrometheusSecret and remove it in our delete code path.

I'm not sure if we should do it before or after we ensure it has the correct content but I'm thinking before we call ensureSecret.

if we do it here then it might have been delete after we validated it.

Reply:
Good catch! I didn't think on that.